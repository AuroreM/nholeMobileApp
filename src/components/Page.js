// @flow

import React, { PropTypes } from 'react';
import { View, StyleSheet, TouchableWithoutFeedback, Keyboard } from 'react-native';

export default (props: PropsType) => (
  <TouchableWithoutFeedback onPress={() => Keyboard.dismiss()}>
    <View
      style={[
        styles.page,
        {
          backgroundColor: props.backgroundColor,
          paddingHorizontal: 24,
          paddingTop: 16,
        },
      ]}
    >
      {props.children}
    </View>
  </TouchableWithoutFeedback>
);

type PropsType = {
  children: React$Element<*> | React$Element<*>[],
  backgroundColor: string,
};

const styles = StyleSheet.create({
  page: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'flex-start',
  },
});
