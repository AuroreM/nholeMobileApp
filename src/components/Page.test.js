import React from 'react';
import 'react-native';
import Page from './Page';
import renderer from 'react-test-renderer';

describe('<Page />', () => {
  it('renders correctly with navBar and margin', () => {
    const tree = renderer.create(<Page backgroundColor={'#fff'} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly without navBar', () => {
    const tree = renderer.create(<Page backgroundColor={'#fff'} noNavBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly without noMargin', () => {
    const tree = renderer.create(<Page backgroundColor={'#fff'} noMargin />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
