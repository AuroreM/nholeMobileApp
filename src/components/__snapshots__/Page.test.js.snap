// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> renders correctly with navBar and margin 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "flex-start",
      },
      Object {
        "backgroundColor": "#fff",
        "paddingHorizontal": 24,
        "paddingTop": 16,
      },
    ]
  }
/>
`;

exports[`<Page /> renders correctly without navBar 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "flex-start",
      },
      Object {
        "backgroundColor": "#fff",
        "paddingHorizontal": 24,
        "paddingTop": 0,
      },
    ]
  }
/>
`;

exports[`<Page /> renders correctly without noMargin 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "flex-start",
      },
      Object {
        "backgroundColor": "#fff",
        "paddingHorizontal": 0,
        "paddingTop": 16,
      },
    ]
  }
/>
`;
